echo/index.js

```diff
@@ -181,13 +181,32 @@ function Client() {
     debug('request options', requestOptions);
 
     request.post(requestOptions, function onResp(err, response, body) {
-      var statusCode = response && response.statusCode
-        ? response.statusCode
-        : 0;
-
-      if (err || parseInt(statusCode / 100) !== 2) {
-        error('[echoClient] addEvents error', { err: err, body: body, statusCode: statusCode });
-        callback(err || 'error response status code: ' + statusCode);
+      if (err) {
+        var errorResponse = {
+          statusCode: 0,
+          message: err.message 
+        };
+        error(
+          '[echoClient] addEvents error', 
+          { 
+            error: errorResponse.message, statusCode: errorResponse.statusCode, body: body 
+          }
+        );
+        callback(errorResponse);
+        return;
+      } else if (response && parseInt(response.statusCode / 100) !== 2) {
+        var errorResponse = {
+          statusCode: response.statusCode,
+          message: response.statusMessage
+        };
+        error(
+          '[echoClient] addEvents error', 
+          { 
+            error: errorResponse.message, statusCode: errorResponse.statusCode, body: body
+          }
+        );
+        callback(errorResponse);
+        return;
```

Comment:
the `return` isn't needed

Reply:
Equally, you could add the `return` here and remove the `else`

Reply:
Not too sure what path to take here, is this more opinion?

Reply:
generally in node we return as early is we can, so `else` is rarely needed.

Reply:
understood